tests/dualstack_test.py

```diff
@@ -128,6 +134,22 @@ def test_dualstack(self) -> None:
                 self.server_xds_host, self.server_xds_port
             )
 
+        with self.subTest("_start_firewall_rule_creation_thread"):
+
+            def recreate_firewall_rule():
```

Comment:
Seems like this functionality is generally useful. Can we just incorporate it into the `--ensure_firewall` flag?